modules/ml/src/main/java/org/apache/ignite/ml/tree/randomforest/RandomForestTrainer.java

```diff
@@ -28,6 +28,8 @@
     protected final int maxDeep;
     /** Min impurity decrease. */
     protected final double minImpurityDecrease;
+    /** Use index structure instead of using sorting while decision tree learning. */
+    protected boolean useIndex = false;
```

Comment:
Somewhere it's final, somewhere it's not. Would be great to use the same approach everywhere. Or you assumed builder approach here?

Reply:
My 5 cents: All trainer's fields in future should be not final due to Parameter Grid feature (it requires setters calls for each tuning field).

Reply:
Wouldn't it be better to create new trainer for each Parameter Grid iteration?